src/plots/linePlot.ts

```diff
@@ -425,6 +449,69 @@ export module Plots {
       return filteredDataIndices;
     }
 
+    private _filterDownsampling(dataset: Dataset, indices: number[]) {
+      let data = dataset.data();
+      let filteredIndices: number[] = [];
+      let min: number;
+      let max: number;
+      let currentSlope: number;
+      let scaledXAcessor = Plot._scaledAccessor(this.x());
+      let scaledYAcessor = Plot._scaledAccessor(this.y());
+
+      if (indices.length === 0) {
+        return filteredIndices;
+      }
+      filteredIndices.push(indices[0]);
+
+      function belongToCurBucket(i: number) {
```

Comment:
Let's not abbreviate the names here

I assume that `Cur` is `Current` but I am unsure.

Also, what are you checking is "belonging to the "Cur" Bucket"?  Is it the index?